rskj-core/src/main/java/co/rsk/peg/storage/BridgeStorageAccessorImpl.java

```diff
@@ -31,19 +31,27 @@ private <T> T getFromRepository(DataWord keyAddress, RepositoryDeserializer<T> d
     }
 
     @Override
-    public <T> void safeSaveToRepository(DataWord addressKey, T object, RepositorySerializer<T> serializer) {
-        try {
-            saveToRepository(addressKey, object, serializer);
-        } catch (IOException ioe) {
-            throw new StorageAccessException("Unable to save to repository: " + addressKey, ioe);
-        }
+    public <T> void safeSaveToRepository(DataWord addressKey, T object, RepositorySerializer<T> serializer) throws IOException {
+        byte[] serializedData = getSerializedData(object, serializer);
+        safeSaveToRepository(addressKey, serializedData);
     }
 
-    private <T> void saveToRepository(DataWord addressKey, T object, RepositorySerializer<T> serializer) throws IOException {
+    private <T> byte[] getSerializedData(T object, RepositorySerializer<T> serializer) throws IOException {
         byte[] data = null;
         if (object != null) {
             data = serializer.serialize(object);
         }
-        repository.addStorageBytes(CONTRACT_ADDRESS, addressKey, data);
+
+        return data;
+    }
+
+    @Override
+    public void safeSaveToRepository(DataWord addressKey, byte[] serializedData) throws IOException {
+        try {
+            repository.addStorageBytes(CONTRACT_ADDRESS, addressKey, serializedData);
+        } catch (RuntimeException e) {
```

Comment:
Let's review this, why was the original method `saveToRepository` throwing an IOException when calling `repository.addStorageBytes`?

Reply:
I wouldn't propagate the IOException here. I think we should catch it and handle it as it was originally

Reply:
you right, my bad. I'll change it to handle it the way it was

Reply:
done